k8s/edge/k8s-template

```diff
@@ -94,36 +89,46 @@ if __name__ == '__main__':
                         help='Command to reload configuration', required=False)
     parser.add_argument('--once', '-n', action='store_true',
                         help='Render template Once', required=False)
-    parser.add_argument('--pause', '-p', help='Pause length, in seconds', type=int,
-                        default=5, required=False)
+    parser.add_argument('--required-label', '-p', help='Pause length, in seconds', type=str,
+                        default='', required=False)
 
     # --------- END OF SECTIONS -----------
     args = parser.parse_args(sys.argv[1:])
 
-    v = vars(args)
-
     if args.verbose or os.getenv('VERBOSE', ''):
         log_level = logging.DEBUG
     else:
         log_level = logging.ERROR
 
     ROOT_LOGGER.setLevel(log_level)
 
-    j2_env = Environment(
-        loader=FileSystemLoader([os.getcwd(), os.path.dirname(os.path.abspath(__file__))])
-    )
-
-    while True:
-        values = find_all_services()
-        with open(args.output, 'w') as output_file:
-            logging.debug('Output file path: %s' % output_file.name)
-            # Reloading template on each iteration, in case it's changed
-            output_file.write(j2_env.get_template(args.template).render({'services': values}))
-
-        if args.command:
-            os.system(args.command)
+    # Reading Namespace from env.vars.
+    namespace = os.environ.get('NAMESPACE', '')
+    if not namespace:
+        raise ValueError("NAMESPACE wasn't found in env var.")
 
-        if args.once:
-            break
-        else:
-            time.sleep(args.pause)
+    # Creating Kubernetes client
+    core_api = client.CoreV1Api(build_client())
+
+    # Reading a list of all services
+    all_services_items = core_api.list_namespaced_service(namespace).items
+    all_services = { service.metadata.name: service for service in all_services_items}
+    render_template(all_services.values(), args.template, args.output, args.command)
+
+    if not args.once:
+        while True:
+            try:
+                w = watch.Watch()
+                for event in w.stream(core_api.list_namespaced_service, namespace, _request_timeout=60):
+                    service = event['object']
+                    event_type = event['type']
+                    if not args.required_label or args.required_label in service.metadata.labels:
+                        service_name = service.metadata.name
```

Comment:
@aliaksandr-d what happen if I remove required label from service that has been registered?

Reply:
@kirillmakhonin *requested label* filters out non-model events. if you remove requested label, it will render *config file* & execute *reload* command on each service change, not only model service.

Reply:
@aliaksandr-d yeah, but it looks like if I remove label from service `args.required_label in service.metadata.labels` would be False and service would not be deleted from `all_services`

Reply:
@kirillmakhonin 
If `required_label` is empty OR `required_label` is in `service.metadata.labels`, then file is regenerated.
In other words, file is regenerated even if you don't specify `required_label`.
If you remove label from service, file won't be regenerated and this service will be omitted due to a condition in *template* file: https://github.com/epam/legion/blob/fb-k8s-service-discovery/k8s/edge/nginx.conf.k8stmpl#L66

Reply:
I've implemented separate *explorer.py* that talks to K8s API